src/OrchardCore.Modules/OrchardCore.Media/Services/MediaFileStoreResolverMiddleware.cs

```diff
@@ -65,12 +63,14 @@ public async Task Invoke(HttpContext context)
             // subpath.Value returns an unescaped path value, subPath returns an escaped path value.
             var subPathValue = subPath.Value;
 
+            var workerKey = context.Request.PathBase.Value + context.Request.Path.Value;
+
```

Comment:
maybe just `context.Request.PathBase.Value + subPathValue`
Or as before if we use a concurrent dictionary per tenant

The middleware is a singleton, so why not having a non static concurrent dictionary per tenant.

Reply:
No reason why not, so updated to use a per instance (i.e. tenant) concurrent dictionary. Probably better as less keys per tenant to lookup